File: 3621-minimum-operations-to-make-array-values-equal-to-k/3621-minimum-operations-to-make-array-values-equal-to-k.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        int minEle = *min_element(nums.begin(), nums.end());
        if(k>minEle) return -1;
        unordered_map<int,int> mp;
        for(auto &num : nums){
            if(num>k){
                mp[num]++;
            }
        }
        return mp.size();
    }
};

File: 3621-minimum-operations-to-make-array-values-equal-to-k/3621-minimum-operations-to-make-array-values-equal-to-k.rs
use std::collections::HashMap;

impl Solution {
    pub fn min_operations(nums: Vec<i32>, k: i32) -> i32 {
        let min_ele = *nums.iter().min().unwrap();
        if k > min_ele {
            return -1;
        }

        let mut mp = HashMap::new();
        for &num in &nums {
            if num > k {
                *mp.entry(num).or_insert(0) += 1;
            }
        }

        mp.len() as i32
    }
}